src/cli/interaction.rs

```diff
@@ -3,24 +3,58 @@ use std::io::{stdin, stdout, BufRead, Write};
 use eyre::{eyre, WrapErr};
 use owo_colors::OwoColorize;
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum PromptChoice {
+    Yes,
+    No,
+    Explain,
+}
+
 // Do not try to get clever!
 //
 // Mac is extremely janky if you `curl $URL | sudo sh` and the TTY may not be set up right.
 // The below method was adopted from Rustup at https://github.com/rust-lang/rustup/blob/3331f34c01474bf216c99a1b1706725708833de1/src/cli/term2.rs#L37
-pub(crate) async fn confirm(question: impl AsRef<str>, default: bool) -> eyre::Result<bool> {
+pub(crate) async fn prompt(
+    question: impl AsRef<str>,
+    default: PromptChoice,
+    explaining: bool,
+) -> eyre::Result<PromptChoice> {
     let stdout = stdout();
     let mut term =
         term::terminfo::TerminfoTerminal::new(stdout).ok_or(eyre!("Couldn't get terminal"))?;
     let with_confirm = format!(
         "\
         {question}\n\
         \n\
-        {are_you_sure} ({yes}/{no}): \
+        {are_you_sure} ({yes}/{no}{maybe_explain}): \
     ",
         question = question.as_ref(),
         are_you_sure = "Proceed?".bold(),
-        no = if default { "n" } else { "N" }.red(),
-        yes = if default { "Y" } else { "y" }.green(),
+        no = if default == PromptChoice::No {
+            "N"
+        } else {
+            "n"
+        }
+        .red(),
+        yes = if default == PromptChoice::Yes {
+            "Y"
+        } else {
+            "n"
+        }
+        .green(),
+        maybe_explain = if !explaining {
```

Comment:
What is this `explaining` bool for? At first I thought it was something like "can this prompt be explained", then I thought it was "are we explaining this prompt", and now I'm not sure.

If it's the first, could it be renamed to `can_explain`, and if it's the second, can it be renamed to `already_explaining` or `is_explaining` or `currently_explaining` or something?